tnkeeh/tnkeeh.py

```diff
@@ -124,11 +125,85 @@ def _remove_long_words(text, threshold=15):
     return (" ").join(word for word in text.split(" ") if len(word) < threshold)
 
 
+def _clean_list(list, segment = False, remove_special_chars = False, 
+        remove_english = False, normalize = False, remove_diacritics = False,
+        execluded_chars = [], remove_tatweel = False, remove_html_elements = False,
+        remove_links = False, remove_twitter_meta = False, remove_long_words = False,
+        remove_repeated_chars = False):
+
+    cleaned_list = []
+
+    for text in list:
+        if segment:
+            # suppress farasa stdout
+            # WARNING: this is LINUX ONLY command!
+            old_stdout = sys.stdout
+            sys.stdout = open(os.devnull, "w")
+            segmenter = FarasaSegmenter(interactive=True)
+            # resume farasa stdout
+            sys.stdout = old_stdout
+
+        if remove_repeated_chars:
+            print('Remove repeated chars')
+            text = _remove_repeated_chars(text)
+        if remove_html_elements:
+            print('Remove HTML elements')
+            text = _remove_html_elements(text)
+        if segment:
+            print('Segment data')
+            text = _farasa_segment(text, segmenter)
+        if remove_english:
+            print('Remove English chars')
+            text = _remove_english_chars(text)
+        if normalize:
+            print('Normalize data')
+            text = _normalize_data(text)
+        if remove_diacritics:
+            print('Remove diacritics')
+            text = _remove_diacritics(text)
+        if remove_special_chars:
+            print('Remove special chars')
+            text = _remove_special_chars(text, execluded_chars)
+        if remove_tatweel:
+            print('Remove tatweel')
+            text = re.sub('ـ', '', text)
+        if remove_links:
+            print('Remove links')
+            text = _remove_links(text)
+        if remove_twitter_meta:
+            print('Remove twitter meta')
+            text = _remove_twitter_meta(text)
+        if remove_long_words:
+            print('Remove long words')
+            text = _remove_long_words(text)
+
+        text = _add_spaces_to_all_special_chars(text)
+        text = _remove_extra_spaces(text)
+        cleaned_list.append(text)
+    return cleaned_list
+
+def clean_data_frame(df, column_name, segment = False, remove_special_chars = False, 
```

Comment:
How to do that more elegantly ? repeating parameters should be done in a better way .

Reply:
@MagedSaeed

Reply:
I think we can pass them as a dict. Here is the idea.
```
default_kwargs = dict(
    segment = False,
    remove_special_chars = False, 
    remove_english = False,
    normalize = False,
    remove_diacritics = False,
    execluded_chars = [],
    remove_tatweel = False,
    remove_html_elements = False,
    remove_links = False,
    remove_twitter_meta = False,
    remove_long_words = False,
    remove_repeated_chars = False
)
```
Then, we can use them in any function call as follows:
`clean_something(something, **default_kwargs)`

Does this sound fine to you?

Reply:
Yes, will give it a try.

Reply:
Seems to work fine. I refactored some of the code.